plugins/azure/resoto_plugin_azure/resource/metrics.py

```diff
@@ -0,0 +1,303 @@
+    average: Optional[float] = field(default=None)
+    total: Optional[float] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AzureMetricTimeSeries:
+    kind: ClassVar[str] = "azure_metric_time_series"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "metadata_values": S("metadatavalues") >> ForallBend(AzureMetricMetadataValues.mapping),
+        "data": S("data") >> ForallBend(AzureMetricTimeSeriesValues.mapping),
+    }
+    metadata_values: Optional[List[AzureMetricMetadataValues]] = field(default=None)
+    data: Optional[List[AzureMetricTimeSeriesValues]] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AzureMetricValue:
+    kind: ClassVar[str] = "azure_metric_value"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "type": S("type"),
+        "name": S("name") >> Bend(AzureMetricValueName.mapping),
+        "displayDescription": S("displayDescription"),
+        "unit": S("unit"),
+        "timeseries": S("timeseries", default=[]) >> ForallBend(AzureMetricTimeSeries.mapping),
+    }
+    id: Optional[str] = field(default=None)
+    type: Optional[str] = field(default=None)
+    name: Optional[str] = field(default=None)
+    display_description: Optional[str] = field(default=None)
+    unit: Optional[str] = field(default=None)
+    timeseries: Optional[List[AzureMetricTimeSeries]] = field(default=None)
+
+
+@define(hash=True, frozen=True)
+class AzureMetricQuery:
+    metric_name: str
+    metric_namespace: str
+    ref_id: str
+    instance_id: str
+    metric_id: str
+    aggregation: str = "average,minimum,maximum"
+    unit: str = "Count"
+
+    @staticmethod
+    def create(
+        metric_name: str,
+        metric_namespace: str,
+        instance_id: str,
+        ref_id: str,
+        metric_id: Optional[str] = None,
+        aggregation: str = "average,minimum,maximum",
+        unit: str = "Count",
+    ) -> "AzureMetricQuery":
+        metric_id = f"{instance_id}/providers/Microsoft.Insights/metrics/{metric_name}"
+        # noinspection PyTypeChecker
+        return AzureMetricQuery(
+            metric_name=metric_name,
+            metric_namespace=metric_namespace,
+            instance_id=instance_id,
+            metric_id=metric_id,
+            aggregation=aggregation,
+            ref_id=ref_id,
+            unit=unit,
+        )
+
+
+@define(eq=False, slots=False)
+class AzureMetricData:
+    kind: ClassVar[str] = "azure_metric"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "timespan": S("timespan"),
+        "interval": S("interval"),
+        "namespace": S("namespace"),
+        "resource_region": S("resourceregion"),
+        "full_metric_values_data": S("value") >> ForallBend(AzureMetricValue.mapping),
+    }
+    full_metric_values_data: List[AzureMetricValue] = field(factory=list)
+    metric_id: Optional[str] = field(default=None)
+    metric_values: Optional[List[float]] = field(default=None)
```

Comment:
- `full_metric_values_data` has the data from azure
- `metric_values` is "invented" and computed by us
correct?

I don't like this model, which depends on the position of the array e.g. `[average, min, max]`.
Can we make it a dict, like:
```python
{
  minimum: 1,
  average: 2,
  maximum: 3
}
```

It would be much clearer.